packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -180,12 +192,29 @@ async function updateWalletAddress(
       update.deactivatedAt = null
     }
 
-    return await walletAddress
-      .$query(deps.knex)
+    const updatedWalletAddress = await walletAddress
+      .$query(trx)
       .patchAndFetch(update)
       .withGraphFetched('asset')
       .throwIfNotFound()
+
+    // Override all existing additional properties if new ones are provided
+    if (additionalProperties && additionalProperties.length) {
```

Comment:
Does nothing if `additionalProperties` is `[]`. Perhaps this is wrong though and an empty array should delete them? Since it's more consistent with how we're otherwise deleting non-existant keys. And because there is currently no way to delete everything. 

I did it this way originally because I thought it would be too easy to accidentally delete everything, but now I see that this reasoning would also apply to deleting non-existent keys.

Reply:
This is a bit of a tricky one. Here is a suggestion:
- `undefined` would ignore doing anything to the additional properties
- `[]` would delete all the additional properties
- keys that match existing values, but have their new value is empty, gets deleted
- new keys with valid values are inserted
- keys with updated values are updated (non-case-sensitive, not sure if this is already expected to the handled by PSQL behaviour?)

Reply:
Agree on all counts except I'd rather _effectively_ update existing keys for the reasons stated here: https://github.com/interledger/rafiki/pull/2769#discussion_r1631403514

What do you think?

Reply:
Agree with both of the last two messages 👍 @koekiebox @BlairCurrey